java/client/src/test/java/glide/api/RedisClientTest.java

```diff
@@ -53,21 +68,239 @@ public void customCommand_interruptedException() throws ExecutionException, Inte
         String key = "testKey";
         Object value = "testValue";
         String cmd = "GETSTRING";
+        String[] arguments = new String[] {cmd, key};
         CompletableFuture<Object> testResponse = mock(CompletableFuture.class);
         InterruptedException interruptedException = new InterruptedException();
         when(testResponse.get()).thenThrow(interruptedException);
-        when(commandManager.submitNewCommand(any(), any())).thenReturn(testResponse);
+
+        // match on protobuf request
+        when(commandManager.submitNewCommand(any(), any(), any(), any())).thenReturn(testResponse);
 
         // exercise
         InterruptedException exception =
                 assertThrows(
                         InterruptedException.class,
                         () -> {
-                            CompletableFuture<Object> response = service.customCommand(new String[] {cmd, key});
+                            CompletableFuture<Object> response = service.customCommand(arguments);
                             response.get();
                         });
 
         // verify
         assertEquals(interruptedException, exception);
     }
+
+    @Test
+    public void ping_success() throws ExecutionException, InterruptedException {
```

Comment:
Consider moving tests to a new file(s)

Reply:
Some have been moved.  But I think we can keep these here until the number of tests gets out of hand.